Src/AutoFixture/DomainNameGenerator.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using Ploeh.AutoFixture.Kernel;
+
+namespace Ploeh.AutoFixture
+{
+    /// <summary>
+    /// Creates new <see cref="DomainName"/> instances.
+    /// </summary>
+    public class DomainNameGenerator : ISpecimenBuilder
+    {
+        private readonly string[] fictitiousDomains =
+        {
+            "example.com",
+            "example.net",
+            "example.org"
+        };
+
+        private readonly Random random = new Random();
+
+        /// <summary>
+        /// Creates a new specimen based on a request.
+        /// </summary>
+        /// <param name="request">The request that describes what to create</param>
+        /// <param name="context">A context that can be used to create other specimens.</param>
+        /// <returns>
+        /// The requested specimen if possible; otherwise a <see cref="NoSpecimen"/> instance.
+        /// </returns>
+        public object Create(object request, ISpecimenContext context)
+        {
+            if (request == null || !typeof(DomainName).Equals(request))
+            {
+                return new NoSpecimen();
```

Comment:
This looks *slightly* incorrect to me.

The convention in AutoFixture is to return `new NoSpecimen(request)` if `request` isn't null.

IIRC, it's actually not used anywhere, but I'd like to be consistent. Also, since AutoFixture is a reusable library, we don't know if someone else relies on this convention, even if AutoFixture itself doesn't. I doubt it, however.

(If I should really do something about this, since it's not used, I ought to remove the overload that takes the request, but that would be a breaking change...)